Grids: Add headerFilter blank item note (#7532)
DevExpress/devextreme-documentation: api-reference/_hidden/GridBaseColumn/lookup/displayExpr.md

[note]

> - Values in a lookup column are sorted by the **valueExpr** field. Implement the column's [calculateSortValue](/api-reference/_hidden/GridBaseColumn/calculateSortValue.md '{basewidgetpath}/Configuration/columns/#calculateSortValue') function if you want to sort by the **displayExpr** field instead:

Reviewer: ```suggestion
- Values in a lookup column are sorted by the **valueExpr** field. Implement the column's [calculateSortValue](/api-reference/_hidden/GridBaseColumn/calculateSortValue.md '{basewidgetpath}/Configuration/columns/#calculateSortValue') function to sort by the **displayExpr** field instead:
```

        <script setup>
        import 'devextreme/dist/css/dx.light.css';
        import Dx{WidgetName}, { DxColumn, DxLookup } from 'devextreme-vue/{widget-name}';

        const calculateSortValue = (data) => {
            const column = this;
            const value = column.calculateCellValue(data);
            return column.lookup.calculateCellValue(value);
        }
        </script>

    ##### React

        <!-- tab: App.js -->
        import React from 'react';
        import 'devextreme/dist/css/dx.light.css';

        import {WidgetName}, {
            Column,
            Lookup
        } from 'devextreme-react/{widget-name}';

        function calculateSortValue (data) {
            const column = this;
            const value = column.calculateCellValue(data);
            return column.lookup.calculateCellValue(value);
        }

        export default function App() {
            return (
                <{WidgetName} ... >
                    <Column ...
                        calculateSortValue={calculateSortValue}>
                        <Lookup ... />
                    </Column>
                </{WidgetName}>
            );
        }

    ---

> - When [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) and **columns**.[lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/) are enabled at the same time, the component adds a `null` value to lookup datasets. To avoid errors, check that items are not `null` before accessing them in the **displayExpr** method:

Reviewer 2: > the component adds a `null` value to lookup datasets

That's not exactly the case. The component adds a null value to the headerFilter datasets, and based on this value, it tries to retrieve the display value from the lookup.

> - When [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) and **columns**.[lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/) are enabled at the same time, the component executes the **displayExpr** method of lookup columns an additional time with a `null` parameter. To avoid errors, check that **data** is not `null` before accessing its properties in the **displayExpr** method:

Reviewer: ```suggestion
- When [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) and **columns**.[lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/) are both enabled, the component also executes the **displayExpr** method of lookup columns with a `null` parameter. To avoid errors, ensure **data** is not `null` before accessing its properties in the **displayExpr** method:
```

Author: @dirkpieterse Upon second thought, it is not very clear what the additional time the displayExpr function is executed with a `null` parameter is in addition to.  I feel to keep this sentence short and understandable, it's best to immediately advise users to check the truthiness of the **data** parameter before accessing its properties:

```suggestion
- When [headerFilter](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/headerFilter/) and **columns**.[lookup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/lookup/) are both enabled, ensure **data** is not `null` before accessing its properties in the **displayExpr** method of lookup columns:
```

Reviewer: Okay
